tests/src/performance/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,355 @@
+                DisplayName = "Execution Time",
+                Unit = "ms"
+            });
+            request.Performance.Metrics.Add(new MetricModel {
+                Name = "ExecutionTime",
+                DisplayName = "Execution Time",
+                Unit = "ms"
+            });
+
+            for (int i = 0; i < s_ScenarioConfiguration.Iterations; ++i)
+            {
+                var startupIteration = new IterationModel { Iteration = new Dictionary<string, double>() };
+                startupIteration.Iteration.Add("ExecutionTime", s_startupTimes[i]);
+                startup.Performance.IterationModels.Add(startupIteration);
+
+                var requestIteration = new IterationModel { Iteration = new Dictionary<string, double>() };
+                requestIteration.Iteration.Add("ExecutionTime", s_requestTimes[i]);
+                request.Performance.IterationModels.Add(requestIteration);
+            }
+
+            return scenarioBenchmark;
+        }
+
+        private static void LaunchProcess(ProcessStartInfo processStartInfo, int timeoutMilliseconds)
+        {
+            PrintRunningStepInformation($"{processStartInfo.FileName} {processStartInfo.Arguments}");
+
+            using (var p = new Process())
+            {
+                p.StartInfo = processStartInfo;
+                p.Start();
+                if (p.WaitForExit(timeoutMilliseconds) == false)
+                {
+                    p.Kill();
+                    throw new TimeoutException($"The process '{processStartInfo.FileName} {processStartInfo.Arguments}' timed out.");
+                }
+
+                if (p.ExitCode != 0)
+                    throw new Exception($"{processStartInfo.FileName} exited with error code {p.ExitCode}");
+            }
+        }
+
+        private static void PrintHeader(string message)
+        {
+            Console.WriteLine();
+            Console.WriteLine("**********************************************************************");
+            Console.WriteLine($"** {message}");
+            Console.WriteLine("**********************************************************************");
+        }
+
+        private static void PrintRunningStepInformation(string message)
+        {
+            Console.WriteLine($"-- {message}");
+        }
+
+        private static readonly ScenarioConfiguration s_ScenarioConfiguration;
+
+        private static int s_iteration;
+        private static double[] s_startupTimes;
+        private static double[] s_requestTimes;
+        private static string s_temporaryDirectory;
+        private static string s_jitBenchDevDirectory;
+
+        /// <summary>
+        /// Provides an interface to parse the command line arguments passed to the JitBench harness.
+        /// </summary>
+        sealed class JitBenchHarnessOptions
+        {
+            public JitBenchHarnessOptions()
+            {
+                _tempDirectory = Directory.GetCurrentDirectory();
```

Comment:
Currently, it would work on the CI, but I agree. It is fragile (e.g. it may break when running depending on the repo path length)!